File: app/src/main/java/com/example/owner/skymood/MyLocationsActivity.java
package com.example.owner.skymood;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.owner.skymood.adapters.MyCardViewAdapter;
import com.example.owner.skymood.model.MyLocation;
import com.example.owner.skymood.model.MyLocationManager;

import java.util.ArrayList;

public class MyLocationsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_my_locations);

        MyLocationManager manager = MyLocationManager.getInstance(this);

        //setting the toolbar
        Toolbar toolbar = (Toolbar) findViewById(R.id.activity_my_locations_view_tool_bar);
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayShowTitleEnabled(false);
        }

        //get my locations
        ArrayList<MyLocation> data = manager.getAllMyLocations();

        //create recycler view
        RecyclerView recycler = (RecyclerView) findViewById(R.id.activity_my_locations_rv_recycler);
        recycler.setLayoutManager(new LinearLayoutManager(this));

        //set my locations data
        MyCardViewAdapter adapter = new MyCardViewAdapter(this, data);
        recycler.setAdapter(adapter);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {

        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        int itemId = item.getItemId();
        if (itemId == R.id.menu_main_item_sky_mood) {
            Intent mainActivity = new Intent(this, MainActivity.class);
            startActivity(mainActivity);
            return true;
        } else if (itemId == R.id.menu_main_item_searched_locations) {
            Intent searchedLocationsActivity = new Intent(this, SearchedLocationsActivity.class);
            startActivity(searchedLocationsActivity);
            return true;
        } else if (itemId == R.id.menu_main_item_my_locations) {
            // do nothing - we are already in this activity
            return true;
        } else {
            return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/com/example/owner/skymood/MyLocationsActivity.kt
package com.example.owner.skymood

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.owner.skymood.adapters.MyCardViewAdapter
import com.example.owner.skymood.model.MyLocationManager

class MyLocationsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_my_locations)

        val manager = MyLocationManager.getInstance(this)

        //setting the toolbar
        val toolbar = findViewById<View?>(R.id.activity_my_locations_view_tool_bar) as Toolbar?
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayShowTitleEnabled(false)

        //get my locations
        val data = manager.allMyLocations

        //create recycler view
        val recycler = findViewById<View?>(R.id.activity_my_locations_rv_recycler) as RecyclerView
        recycler.setLayoutManager(LinearLayoutManager(this))

        //set my locations data
        val adapter = MyCardViewAdapter(this, data)
        recycler.setAdapter(adapter)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem) = when (item.itemId) {
        R.id.menu_main_item_sky_mood -> {
            val mainActivity = Intent(this, MainActivity::class.java)
            startActivity(mainActivity)
            true
        }

        R.id.menu_main_item_searched_locations -> {
            val searchedLocationsActivity = Intent(this, SearchedLocationsActivity::class.java)
            startActivity(searchedLocationsActivity)
            true
        }

        R.id.menu_main_item_my_locations -> {
            // do nothing - we are already in this activity
            true
        }

        else -> super.onOptionsItemSelected(item)
    }
}
